src/lib/signature_kind/mina_signature_kind.mli

```diff
@@ -3,6 +3,20 @@ type t = Mina_signature_kind_type.t =
   | Mainnet
   | Other_network of string
 
+val bin_t : t Bin_prot.Type_class.t
```

Comment:
`bin_io_unversioned` can't be used in signature environment.

I believe many codes in our codebase have this hand-unrolling issue.